source/material_model/non_dimensional.cc

```diff
@@ -0,0 +1,194 @@
+/*
+  Copyright (C) 2016 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/material_model/non_dimensional.h>
+
+using namespace dealii;
+
+namespace aspect
+{
+  namespace MaterialModel
+  {
+    template <int dim>
+    void
+    Nondimensional<dim>::
+    evaluate(const MaterialModel::MaterialModelInputs<dim> &in,
+             MaterialModel::MaterialModelOutputs<dim> &out) const
+    {
+      for (unsigned int i=0; i < in.temperature.size(); ++i)
+        {
+          const Point<dim> position = in.position[i];
+          const double temperature_deviation = in.temperature[i] - this->get_adiabatic_conditions().temperature(position);
+          const double pressure_deviation = in.pressure[i] - this->get_adiabatic_conditions().pressure(position);
+
+          const double depth = this->get_geometry_model().depth(position);
+
+          out.viscosities[i] = (compressible ? Di : 1.0) / Ra
+                               * exp(-b*temperature_deviation + c*depth);
+
+          out.specific_heat[i] = reference_specific_heat;
+          out.thermal_conductivities[i] = 1.0;
+          out.thermal_expansion_coefficients[i] = compressible ? Di : 1.0;
+
+          out.densities[i] = reference_rho
+                             * exp(depth * Di/gamma)
+                             * (1.0
+                                - out.thermal_expansion_coefficients[i] * temperature_deviation
+                                + (tala?0.0:1.0)*Di*gamma * pressure_deviation );
+
+          out.compressibilities[i] = 0.0;
```

Comment:
We should add an Assert that this material model can not be used with `isothermal_compression`. This line would cause some confusion (it creates and incompressible model in that approximation).